toolchain/sem_ir/typed_insts.h

```diff
@@ -793,6 +793,24 @@ struct ImplDecl {
   DeclInstBlockId decl_block_id;
 };
 
+// The type of an associated function within an `impl`. This is the type of
+// `(SelfType as Interface).AssociatedFunction`.
+struct ImplFunctionType {
+  static constexpr auto Kind =
+      InstKind::ImplFunctionType.Define<Parse::NoneNodeId>(
+          {.ir_name = "impl_fn_type",
+           .is_type = InstIsType::Always,
+           .constant_kind = InstConstantKind::Conditional,
+           .is_lowered = false});
+
+  TypeId type_id;
+  // The type of the function within the interface. This includes the
```

Comment:
StringifyTypeExpr makes it look like this will always be a `FunctionType`, is that correct?
```suggestion
  // The `FunctionType` within the interface. This includes the
```